diff --git a/configure b/configure
index ecc8bdf..793f76c 100755
--- a/configure
+++ b/configure
@@ -248,7 +248,6 @@ ARCH_EXT_LIST="
 HAVE_LIST="
     ${ARCH_EXT_LIST}
     vpx_ports
-    pthread_h
     unistd_h
 "
 EXPERIMENT_LIST="
@@ -559,7 +558,6 @@ process_detect() {
 int main(void) {return 0;}
 EOF
     # check system headers
-    check_header pthread.h
     check_header unistd.h # for sysconf(3) and friends.
 
     check_header vpx/vpx_integer.h -I${source_path} && enable_feature vpx_ports
diff --git a/vp8/common/generic/systemdependent.c b/vp8/common/generic/systemdependent.c
index 28c981a..291397f 100644
--- a/vp8/common/generic/systemdependent.c
+++ b/vp8/common/generic/systemdependent.c
@@ -21,7 +21,8 @@
 #if CONFIG_MULTITHREAD
 #if HAVE_UNISTD_H && !defined(__OS2__)
 #include <unistd.h>
-#elif defined(_WIN32)
+#endif
+#if defined(_WIN32)
 #include <windows.h>
 typedef void(WINAPI *PGNSI)(LPSYSTEM_INFO);
 #elif defined(__OS2__)
